splunk_logger.go

```diff
@@ -517,6 +524,73 @@ func (l *splunkLogger) queueMessageAsync(message *splunkMessage) error {
 	return nil
 }
 
+func telemetry(info logger.Info, l *splunkLogger, sourceType string, splunkFormat string) {
+
+	//Send weekly
+	timer := time.NewTicker(600000000000000)
+	messageArray := []*splunkMessage{}
+	timestamp := strconv.FormatInt(time.Now().UTC().UnixNano()/int64(time.Second), 10)
+
+	type telemetryEvent struct {
+		Component string `json:"component"`
+		Type      string `json:"type"`
+		Data      struct {
+			JsonLogs                bool   `json:"jsonLogs"`
+			PartialMsgBufferMaximum int    `json:"partialMsgBufferMaximum"`
+			PostMessagesBatchSize   int    `json:"postMessagesBatchSize"`
+			PostMessagesFrequency   int    `json:"postMessagesFrequency"`
+			SplunkFormat            string `json:"splunkFormat"`
+			StreamChannelSize       int    `json:"streamChannelSize"`
+			Sourcetype              string `json:"sourcetype"`
+		} `json:"data"`
+		OptInRequired int64 `json:"optInRequired"`
+	}
+
+	telem := &telemetryEvent{}
+	telem.Component = "app.connect.docker"
+	telem.Type = "event"
+	telem.OptInRequired = 2
+	telem.Data.Sourcetype = sourceType
+	telem.Data.SplunkFormat = splunkFormat
+	telem.Data.PostMessagesFrequency = int(getAdvancedOptionDuration(envVarPostMessagesFrequency, defaultPostMessagesFrequency))
+	telem.Data.PostMessagesBatchSize = getAdvancedOptionInt(envVarPostMessagesBatchSize, defaultPostMessagesBatchSize)
+	telem.Data.StreamChannelSize = getAdvancedOptionInt(envVarStreamChannelSize, defaultStreamChannelSize)
+	telem.Data.PartialMsgBufferMaximum = getAdvancedOptionInt(envVarBufferMaximum, defaultBufferMaximum)
```

Comment:
Are these advanced options needed for a 1 shot telemetry event - is this bloat?

Reply:
This is the config snapshot as discussed